### ISSUE TITLE
Model unable to support batching on triton inference server

### ISSUE THREAD

**Issue by benhgm** [triton-inference-server/server#5866]
**Description**
I am getting an error (screenshot below) when attempting to enable batching on Triton Inference Server.

![image](https://github.com/triton-inference-server/server/assets/80461537/b28f3df1-6b17-4a9a-b2c7-7001d269f9d1)

**Triton Information**
21.08 (nvcr.io/nvidia/tritonserver:21.08-py3)

Are you using the Triton container or did you build it yourself?

**To Reproduce**
Model: Pytorch-based model (Resnet-50) + custom classification head, converted to ONNX using the following python code.
```
model = get_model('resnet50_nfc', 22)
model.load_state_dict(torch.load('models/peta/resnet50_new/model.pth'))
model.eval()

trace_input = torch.randn(8, 3, 288, 144)
dummy_output = model(trace_input)

output_names = ['input.1', '512']
dyn_axes = {k: {0: "batch_size"} for k in output_names}

torch.onnx.export(
        model,
        trace_input,
        "model.onnx",
        verbose=True, 
        dynamic_axes=dyn_axes,
        opset_version=11)
```

Config file: please see below
```
name: "par"
backend: "onnxruntime"
max_batch_size: 16
input [
    {
        name: "input.1"
        data_type: TYPE_FP32
        dims: [3, 288, 144]
    }
]
output [
    {
        name: "512"
        data_type: TYPE_FP32
        dims: [22]
    }
]

dynamic_batching {}
```

Model Architecture: Resnet-50 backbone + Linear + BatchNorm + LeakyReLU + Sigmoid Head

**Expected behavior**
Model to be served with no errors and dynamic batching enabled

**Comment by rmccorm4** [comment]
Hi @benhgm,

This looks like a model issue. From Triton's perspective, it's likely just an error in the batching/dynamic shape logic of the ONNX model itself.

I would take a look at your model with a tool like [Netron](https://netron.app/) and see if the model I/O shapes look correct or not around the problem nodes/ops in your ONNX model.

Also from a glance at your code snippet, you may need to include both input and output names in your `dyn_axes`. It looks like you're not including the output name "512" in your snippet above, which may be the root cause of the issue. Here's a similar dynamic torch->onnx conversion script: https://gist.github.com/rmccorm4/b72abac18aed6be4c1725db18eba4930#file-alexnet_onnx-py-L20 for reference.

**Comment by benhgm** [comment]
Hi @rmccorm4 thanks for the advice. After going through my conversion script again, I found that the issue was that i did not send my model to the GPU using model.cuda() before doing the conversion. After I implemented model.cuda(), I was able to switch on dynamic batching succesfully

**Comment by rmccorm4** [comment]
Glad you got it working! I'll go ahead and close the issue then.
